src/main/java/com/theelixrlabs/healthcare/service/PatchDoctorService.java

```diff
@@ -0,0 +1,98 @@
+package com.theelixrlabs.healthcare.service;
+
+import com.theelixrlabs.healthcare.constants.DoctorConstants;
+import com.theelixrlabs.healthcare.dto.DoctorDto;
+import com.theelixrlabs.healthcare.exceptionHandler.CustomException;
+import com.theelixrlabs.healthcare.model.DoctorModel;
+import com.theelixrlabs.healthcare.repository.DoctorRepository;
+import org.springframework.stereotype.Service;
+
+import java.util.Optional;
+import java.util.UUID;
+
+/**
+ * Service class for facilitating partial updates to doctor details.
+ */
+@Service
+public class PatchDoctorService {
+
+    private final DoctorRepository doctorRepository;
+
+    public PatchDoctorService(DoctorRepository doctorRepository) {
+        this.doctorRepository = doctorRepository;
+    }
+
+    /**
+     * Replace existing doctor details based on request.
+     *
+     * @param doctorId  Doctor unique UUID
+     * @param doctorDto Data transfer object containing doctor information.
+     * @return The replaced doctor details.
+     * @throws CustomException If validation fails or doctor already exists.
+     */
+    public DoctorDto patchDoctorById(UUID doctorId, DoctorDto doctorDto) throws CustomException {
+        Optional<DoctorModel> optionalDoctor = doctorRepository.findById(doctorId);
+        validateDoctor(doctorDto);
+        if (optionalDoctor.isPresent()) {
+            DoctorModel existingDoctor = optionalDoctor.get();
+            if (doctorDto.getFirstName() != null) {
+                existingDoctor.setFirstName(doctorDto.getFirstName());
+            }
+            if (doctorDto.getLastName() != null) {
+                existingDoctor.setLastName(doctorDto.getLastName());
+            }
+            if (doctorDto.getDepartment() != null) {
+                existingDoctor.setDepartment(doctorDto.getDepartment());
+            }
+            if (doctorDto.getAadhaarNumber() != null) {
+                String aadhaarNumber = doctorDto.getAadhaarNumber();
+                String formattedAadhaarNumber = aadhaarNumber.substring(0, 4) + " " +
```

Comment:
move these empty spaces " " to DoctorConstants

Reply:
added to DoctorConstant class